lib/connections/structs.go

```diff
@@ -165,3 +166,19 @@ func (o *onAddressesChangedNotifier) notifyAddressesChanged(l genericListener) {
 		callback(l)
 	}
 }
+
+type dialTarget struct {
+	dialer   genericDialer
+	uri      *url.URL
+	priority int
+}
+
+func (t dialTarget) Dial(deviceID protocol.DeviceID, wg sync.WaitGroup, res chan<- internalConn) {
+	l.Debugln("dialing", deviceID, t.uri, "prio", t.priority)
+	conn, err := t.dialer.Dial(deviceID, t.uri)
+	l.Debugln("dialing", deviceID, t.uri, "outcome", conn, err)
+	if err == nil {
+		res <- conn
```

Comment:
How about, instead of having `res` be `len(targets)`-buffered, it is just 1-buffered and we do

```
select {
  case res <- conn:
    // we won
  default:
    // we lost
    conn.close()
}
```

then I think you don't need the discarder routine above?

Reply:
Not really. It will panic on send to closed channel, and second dial can slip through to the channel which still has to be discarded.